WordPress/gutenberg
# WIP: Try new api for in toolbar editing (and fix image block toolbar focus loss issue).

Note: this PR currently includes the reversion changes from #25854. Ideally that will be merged first before this is code reviewed.

## Description

Fixes #24676, Fixes #23721, related #23375

Using the image block's crop tool from the keyboard resulted in a focus loss.

This PR fixes the issue, introducing a new in-toolbar editing system.

The system works like this:
- A new slot/fill is introduced `__experimentalBlockToolbarInlineEdit`.
- This slot is rendered in the `BlockPopover` as a sibling to the `BlockContextualToolbar`
- When a fill is rendered in this slot, an action is triggered that visually hides the block contextual toolbar (applying a classname to it) and the fill is displayed in a `PopoverWrapper`.

The advantage of this system compared to #23613 is that it doesn't require as many changes to the toolbar, the only change is an additional classname that hides the toolbar.

It also follows the advice provided by the @enriquesanchez and @afercia in using a popover for the content that 'replaces' the toolbar.

TODO: 
- [x] Improve styling
- [ ] Come up with a way to get the top toolbar working
- [ ] Improve DevEx, hide some of the implementation, possibly wrap this up in a component similar to `<Dropdown>`
- [ ] Ensure any new APIs are experimental

## How has this been tested?
Prerequisite: Ensure top toolbar is not active (this doesn't work yet for the top toolbar).
1. Add an image block and upload or choose an image from the media library
2. Shift tab to the toolbar and use the right arrow key to move to the crop button
3. Use space/enter to select the crop button, observe that crop tools are shown with focus moved to the first button
4. Press escape, observe that the crop tools are hidden and focus moves back to the crop button

Other things to test
- Cancel and Apply buttons work correctly
- Moving focus outside the crop tools should close them

## Screenshots 

## Types of changes

Bug fix (non-breaking change which fixes an issue)



## Checklist:
